Apps/W1/Shopify/app/src/Base/Codeunits/ShpfyCommunicationMgt.Codeunit.al

```diff
@@ -116,18 +116,23 @@ codeunit 30103 "Shpfy Communication Mgt."
     var
         ShpfyGraphQLRateLimit: Codeunit "Shpfy GraphQL Rate Limit";
         ShpfyJsonHelper: Codeunit "Shpfy Json Helper";
+        ReceivedData: Text;
         ErrorOnShopifyErr: Label 'Error(s) on Shopify:\ \%1', Comment = '%1 = Errors from json structure.';
+        NoJsonErr: Label 'The response from Shopify contains no JSON. \Requested: %1 \Response: %2', Comment = '%1 = The request = %2 = Receibve data';
     begin
         ShpfyGraphQLRateLimit.WaitForRequestAvailable(ExpectedCost);
-        if JResponse.ReadFrom(ExecuteWebRequest(CreateWebRequestURL('graphql.json'), 'POST', GraphQLQuery, ResponseHeaders, 0)) then
+        ReceivedData := ExecuteWebRequest(CreateWebRequestURL('graphql.json'), 'POST', GraphQLQuery, ResponseHeaders, 0);
+        if JResponse.ReadFrom(ReceivedData) then begin
```

Comment:
Do you think it could improve code readability if the condition is reversed?

if not JResponse.ReadFrom(ReceivedData) then
    Error(...);

// Rest of the code goes here.